test/e2e/istio_test.go

```diff
@@ -0,0 +1,212 @@
+	"github.com/stretchr/testify/suite"
+)
+
+type IstioTestSuite struct {
+	suite.Suite
+}
+
+// LIFECYCLE - Suite
+func (suite *IstioTestSuite) SetupSuite() {
+	t = suite.T()
+	var err error
+	ctx, err = prepare(t)
+	if err != nil {
+		if ctx != nil {
+			ctx.Cleanup()
+		}
+		require.FailNow(t, "Failed in prepare")
+	}
+	fw = framework.Global
+	namespace = ctx.GetID()
+	require.NotNil(t, namespace, "GetID failed")
+
+	// label namespace
+	ns, err := framework.Global.KubeClient.CoreV1().Namespaces().Get(context.Background(), namespace, metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("failed to get the namespaces details: %v", err)
+	}
+	labels := ns.GetLabels()
+
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+
+	labels["istio-injection"] = "enabled"
+
+	ns.SetLabels(labels)
+
+	ns, err = framework.Global.KubeClient.CoreV1().Namespaces().Update(context.Background(), ns, metav1.UpdateOptions{})
+
+	if err != nil {
+		t.Errorf("failed to update labels of the namespace details: %v", err)
+	}
+
+	addToFrameworkSchemeForSmokeTests(t)
+}
+
+func (suite *IstioTestSuite) TearDownSuite() {
+	handleSuiteTearDown()
+}
+
+// LIFECYCLE - Test
+
+func (suite *IstioTestSuite) SetupTest() {
+	t = suite.T()
+}
+
+func (suite *IstioTestSuite) AfterTest(suiteName, testName string) {
+	handleTestFailure()
+}
+
+func TestIstioSuite(t *testing.T) {
+	suite.Run(t, new(IstioTestSuite))
+}
+
+func (suite *IstioTestSuite) TestEnvoySidecar() {
+	// First deploy a Jaeger instance
+	jaegerInstanceName := "simplest"
+	jaegerInstance := createJaegerInstanceFromFile(jaegerInstanceName, "../../examples/simplest.yaml")
+	defer undeployJaegerInstance(jaegerInstance)
+	err := WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName, 1, retryInterval, timeout+(1*time.Minute))
+	require.NoError(t, err)
+
+	// Now deploy examples/business-application-injected-sidecar.yaml
+	businessAppCR := getBusinessAppCR(err)
+	defer os.Remove(businessAppCR.Name())
+	cmd := exec.Command("kubectl", "create", "--namespace", namespace, "--filename", businessAppCR.Name())
+	output, err := cmd.CombinedOutput()
+	if err != nil && !strings.Contains(string(output), "AlreadyExists") {
+		require.NoError(t, err, "Failed creating Jaeger instance with: [%s]\n", string(output))
+	}
+	const vertxDeploymentName = "myapp"
+	err = WaitForDeployment(t, fw.KubeClient, namespace, vertxDeploymentName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Failed waiting for myapp deployment")
+
+	// Add a liveliness probe to create some traces
+	vertxPort := intstr.IntOrString{IntVal: 8080}
+	livelinessHandler := &corev1.HTTPGetAction{Path: "/", Port: vertxPort, Scheme: corev1.URISchemeHTTP}
+	handler := &corev1.Handler{HTTPGet: livelinessHandler}
+	livelinessProbe := &corev1.Probe{Handler: *handler, InitialDelaySeconds: 1, FailureThreshold: 3, PeriodSeconds: 10, SuccessThreshold: 1, TimeoutSeconds: 1}
+
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
```

Comment:
Can you refactor this whole wait.Poll block and move it to utils.go as a function named WaitForVertxUpdate() or something similar?  If you have time it would be great if you could also apply this change to the examples1_test.go or open an issue noting it needs to be done.

Reply:
Done. Pls have a review.